docs: docs for new KEY_FORMAT property and associated server configs (#6252)
confluentinc/ksql: docs/operate-and-deploy/installation/server-config/config-reference.md

[ksqlDB Server image](https://hub.docker.com/r/confluentinc/ksqldb-server/)
is `KSQL_KSQL_FUNCTIONS_SUBSTRING_LEGACY_ARGS`.

### ksql.persistence.default.format.key

Sets the default value for the `KEY_FORMAT` property if one is
not supplied explicitly in [CREATE TABLE](../../../developer-guide/ksqldb-reference/create-table.md)
or [CREATE STREAM](../../../developer-guide/ksqldb-reference/create-stream.md)
statements.

> The default value for this configuration is `KAFKA`.

Author: I'll add this for consistency but it's not actually possible for there to be no default key format right now, since the default server config contains a default and there's no way to override that with "no default" currently.

For supported formats, see [Serialization Formats](../../../developer-guide/serialization.md#serialization-formats).

### ksql.persistence.default.format.value

Sets the default value for the `VALUE_FORMAT` property if one is
not supplied explicitly in [CREATE TABLE](../../../developer-guide/ksqldb-reference/create-table.md)
or [CREATE STREAM](../../../developer-guide/ksqldb-reference/create-stream.md)
statements.

> If not set and no explicit value is provided in the statement, via either the `VALUE_FORMAT` or the
> `FORMAT` property, an error will be thrown.

Reviewer: For user facing docs, try not to talk in programming terms e.g. 'and error will be thrown'. (We all do it!)

```suggestion
If not set and no explicit value is provided in the statement, via either the `VALUE_FORMAT` or the
`FORMAT` property, the statement will be rejected as invalid.
```

Author: Good tip. Thanks!
